tests/floating-point/fft/fft-comb-pease.fil

```diff
@@ -0,0 +1,124 @@
+import "primitives/signed.fil";
+import "primitives/math/complex.fil";
+
+/// Implements a combinational PEASE fft dataflow.
+comp CombPeaseFFT[NStages, W, E, ?M=W-E-1]<'G: 1>(
+  in[NPoints*2]: ['G, 'G+1] W
+) -> (
+  out[NPoints*2]: ['G, 'G+1] W
+) with {
+  let NPoints = pow2(NStages);
+} where
+  NStages > 0,
+  W == E + M + 1,
+  E > 1,
+  M > 1
+{
+  assume pow2(NStages-1) == NPoints/2;
+  assume pow2(NStages-2) == NPoints/4;
+
+  // generate twiddle factors
+  bundle twiddle[NPoints]: ['G, 'G+1] W;
+  for i in 0..NPoints/2 {
+    // assume i*2 < NPoints;
+    cosbits := new SignedConst[W, cos_bits(i, NPoints)]<'G>();
+    sinbits := new SignedConst[W, sin_bits(i+pow2(NStages-1), NPoints)]<'G>(); // equivalent to -sin
+    twiddle{i*2} = cosbits.out;
+    twiddle{i*2+1} = sinbits.out;
+  }
+
+  // NPoints*2 * (Nstages+1) bus storing the inputs and outputs for every stage
+  bundle data[NPoints*2 * (NStages+1)]: ['G, 'G+1] W;
+  // bit reverse the input
+  for j in 0..NPoints {
+    let br = bit_rev(j, NStages);
+    assume br < NPoints;
+    data{j*2..j*2+2} = in{br*2..br*2+2};
+  }
+
+  // perform the fft
+  for stage in 0..NStages {
+    stage_twiddle := new TwiddleStage[NStages, stage, W]<'G>(twiddle{0..NPoints});
+
+    let data_offset = NPoints * stage * 2;
+
+    // temporary data storage before stride permutation
+    bundle tmp_data[NPoints*2]: ['G, 'G+1] W;
+
+    // perform the butterflies
+    for butterfly_i in 0..pow2(NStages-1) {
+      // needed to speed up the solver
+      assume butterfly_i < NPoints/2;
```

Comment:
Out of curiosity: how did you figure out that this was the thing that was needed?

Reply:
I realized after commenting out other sections of the code that this was nearly freezing `cvc5`, and I added this to make it work. I think it's actually not necessary now because we have that `pow2(NStages-1) == NPoints/2` further above.

Reply:
Cvc5 has been consistently slower so we should replace it with z3 as the default solver